### ISSUE TITLE
Consider swapping the "Pin" and "Edit" message options

### ISSUE THREAD

**Issue by babolivier** [element-hq/element-x-android#4046]
#### What would you like to do?

Having easy access to message features I use most often.

#### Why would you like to do it?

I edit messages way more than I pin them. Therefore I believe it makes more sense for the "Edit" option to be close to the top of the message context menu, and "Pin" close to the bottom. A recent update moved the "Pin" option to the 3rd position, and "Edit" to the 5th, meaning I now need to scroll to access one of the features that is the most useful to me.

Moreover, it seems my muscle memory tends to favour somewhere around the 3rd position in that menu; my memory isn't great so I'm not sure if a previous version of Element X had that option there, or if this is caused by something else. But at least in one occurrence, recently, I've been in a rush to edit a message, and I ended up pinning it by mistake. And, just this morning, I've seen the same thing happen to a friend I was chatting with.

#### How would you like to achieve it?

Swapping the "Pin" And "Edit" options in the message context menu.

### Are you willing to provide a PR?

No

**Comment by bmarty** [comment]
@americanrefugee what do you think about this suggestion? I tend to agree with it. Please confirm before we take any action.

Order is taking from https://www.figma.com/design/ux3tYoZV9WghC7hHT9Fhk0/Compound-iOS-Components?node-id=2946-2392 and in the code [here](https://github.com/element-hq/element-x-android/blob/develop/features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/actionlist/model/TimelineItemActionComparator.kt#L10).

CC @mxandreas

**Comment by mxandreas** [comment]
I think the actions that are more frequently used, should be closer to your thumb - which is at the bottom. This seems to be a common design practice in other apps as well.

However, if the list of actions is now that large that, the bottom of the list is not visible and requires scrolling, then following that principle is not possible. Thus, I think we should try to fix this - e.g. can we get rid of the need to scroll to the bottom?

**Comment by americanrefugee** [comment]
@bmarty I accept the proposal and have changed the ordering according to what was suggested:
- [iOS](https://www.figma.com/design/ux3tYoZV9WghC7hHT9Fhk0/Compound-iOS-Components?node-id=2946-2134&t=0haRXC4C9iTSpXkp-4)
- [Android](https://www.figma.com/design/G1xy0HDZKJf5TCRFmKb5d5/Compound-Android-Components?node-id=2736-1252&t=TA0YzEdDlKjefFYg-4)

@mxandreas @bmarty I never intended for the Share sheet to be scrollable in most cases. The number of list items is dependent on the user type and permissions. So, the size of the list will change, and the size of the screen will determine how much of the list we can show at once. In the worst case scenario, where all possible options are visible, the sheet might need minimal scrolling on a device screen that's only 800 px. But in the vast majority of cases, the list should not scroll.

**Comment by mxandreas** [comment]
@americanrefugee Please have look at my previous comment and confirm if you really want to change the order.

**Comment by americanrefugee** [comment]
I did look and still think this is the right thing.

- According to your logic, we would need to reorder the entire list.
- Having the least used options at the top definitely feels strange to me, and without knowing how many instances the list will scroll, it seems dangerous.
- But most importantly, regardless of what other apps do, we've trained our users to find the most important options at the top. Like @babolivier mentioned, our users get very frustrated when we break their muscle memory.

So, either we go with the above OR I think we need to align internally (Design and Product) before reconsidering the position of all the options.
